### ISSUE TITLE
osu!mania 4k/7k ranking / pp values are not present on profile overlay

### ISSUE THREAD

**Issue by peppy** [ppy/osu#30827]
I guess also on the user card somewhere for the local user?

### Discussed in https://github.com/ppy/osu/discussions/30787

<div type='discussions-op-text'>

<sup>Originally posted by **theredpro-sj** November 20, 2024</sup>
I'm not very confident in how I worded the title but the basic idea of it is that on the website when hovering global rank in the mania gamemode you also see the users 4k and 7k rank along with their peak rank like in other gamemodes (see first image). This behavior is not replicated on the lazer client as when hovering over global rank it only shows peak rank and doesn't show the 4k and 7k ranks (see second image). Basically I feel that the 4k and 7k breakdown should be shown when hovering over a users global rank like it is on the website.
<img width="513" alt="Screenshot 2024-11-19 at 6 12 39 PM" src="https://github.com/user-attachments/assets/6b9ae797-5b2b-449a-96f9-9373172e2d3b">

<img width="1470" alt="Screenshot 2024-11-19 at 6 11 39 PM" src="https://github.com/user-attachments/assets/3396f52d-4619-4864-b983-434d2d1dfe4e">

</div>

**Comment by theredpro-sj** [comment]
In an extension to this there is no way to separate the 4k/7k ranking on the ingame leaderboard.
<img width="1470" alt="Screenshot 2024-11-22 at 1 30 21 PM" src="https://github.com/user-attachments/assets/7e944dbe-f083-421e-9894-68132876775d">

**Comment by mESmaC** [comment]
I've started looking into this issue and noticed that the UserRankPanel (lines 187-200) contains a TODO for the tooltip, which seems relevant. Adding a breakpoint to the encompassing method confirms it triggers when navigating to the relevant screen. However, I couldn’t find any existing user strings in the UserStrings file for the 4k/7k-specific statistics.

I’ll investigate further to determine how to implement this, but I wanted to confirm: should new strings be added to UserStrings for these statistics, or is there a preferred way to structure this data in the tooltip?

**Comment by frenzibyte** [comment]
`UserRankPanel` is one of many places where this should be supported. Check out all the above screenshots and locate the relevant components and hook up the displays with the values returned from the API.

**Comment by chadmoyashii** [comment]
Hello! New dev here. Decided to tackle this issue for a school assignment, however, hit a similar roadblock to what was written above. (I couldn't get the user strings for the 4k/7k statistics) I unfortunately didn't get the code to work, however, I have uploaded my work to my personal repo in case you would like to reference some of the ideas I had. (I also didn't want to create an incomplete PR in this project to waste anyone's time) 

https://github.com/chadmoyashii/osu4k7kdisplayissue

Feel free to take a look at it, maybe you'll find some inspiration for when you resolve this issue!

Cheers

**Comment by NicholasChin28** [comment]
Hello! First time contributor here. Decided to tackle this issue after noticing similar behavior between the client and web implementation.
After comparing the code, I found that the web version handles the 4K/7K statistics through a Variant mapping in the API response:

```typescript
interface Variant {
    country_rank: number | null;
    global_rank: number | null;
    mode: Ruleset;
    pp: number;
    variant: '4k' | '7k';
}
```
Source: [user-statistics-json.ts](https://github.com/ppy/osu-web/blob/master/resources/js/interfaces/user-statistics-json.ts)
I've implemented a similar mapping in the `UserStatistics` class and created a pull request [#31084](https://github.com/ppy/osu/pull/31084)
 with the changes. Testing shows the tooltips now display the 4K/7K rankings correctly.
Feel free to check it out and provide any feedback! Happy to make adjustments or have the implementation serve as reference for future solutions!
